File: 0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        map<char, int> hashbrown;
        for (auto m : magazine) {
            hashbrown[m]++;
        }
        
        for (auto r : ransomNote) {
            if (hashbrown[r] == 0) {
                return false;
            }
            hashbrown[r]--;
        }
        
        return true;
    }
};

File: 0383-ransom-note/0383-ransom-note.java
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        hashbrown = {}
        for m in magazine:
            hashbrown[m] = hashbrown.get(m, 0) + 1 # get() provides a default val
         
        for r in ransomNote:
            if hashbrown.get(r, 0) == 0:
                return False
            hashbrown[r] -= 1
            
        return True
        